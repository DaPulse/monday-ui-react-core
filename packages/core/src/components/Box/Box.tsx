import React, { forwardRef, useRef } from "react";
import cx from "classnames";
import useMergeRef from "../../hooks/useMergeRef";
import {
  BackgroundColor as BackgroundColorEnum,
  Border as BorderEnum,
  BorderColor as BorderColorEnum,
  BoxTextColor as BoxTextColorEnum,
  Margin as MarginEnum,
  MarginBottom as MarginBottomEnum,
  MarginEnd as MarginEndEnum,
  MarginStart as MarginStartEnum,
  MarginTop as MarginTopEnum,
  MarginX as MarginXEnum,
  MarginY as MarginYEnum,
  Padding as PaddingEnum,
  PaddingBottom as PaddingBottomEnum,
  PaddingEnd as PaddingEndEnum,
  PaddingStart as PaddingStartEnum,
  PaddingTop as PaddingTopEnum,
  PaddingX as PaddingXEnum,
  PaddingY as PaddingYEnum,
  Rounded as RoundedEnum,
  Shadow as ShadowEnum
} from "./BoxConstants";
import {
  BackgroundColor,
  Border,
  BorderColor,
  BoxTextColor,
  Margin,
  MarginBottom,
  MarginEnd,
  MarginStart,
  MarginTop,
  MarginX,
  MarginY,
  Padding,
  PaddingBottom,
  PaddingEnd,
  PaddingStart,
  PaddingTop,
  PaddingX,
  PaddingY,
  Rounded,
  Shadow
} from "./Box.types";
import { VibeComponent, VibeComponentProps, withStaticProps, ElementContent } from "../../types";
import styles from "./Box.module.scss";

export interface BoxProps extends VibeComponentProps {
  elementType?: keyof JSX.IntrinsicElements | string;
  children?: ElementContent;
  disabled?: boolean;
  border?: Border;
  borderColor?: BorderColor;
  rounded?: Rounded;
  shadow?: Shadow;
  margin?: Margin;
  marginX?: MarginX;
  marginY?: MarginY;
  marginTop?: MarginTop;
  marginEnd?: MarginEnd;
  marginBottom?: MarginBottom;
  marginStart?: MarginStart;
  padding?: Padding;
  paddingX?: PaddingX;
  paddingY?: PaddingY;
  paddingTop?: PaddingTop;
  paddingEnd?: PaddingEnd;
  paddingBottom?: PaddingBottom;
  paddingStart?: PaddingStart;
  backgroundColor?: BackgroundColor;
  textColor?: BoxTextColor;
  scrollable?: boolean;
  style?: React.CSSProperties;
}

const Box: VibeComponent<BoxProps> & {
  borders?: typeof BorderEnum;
  borderColors?: typeof BorderColorEnum;
  roundeds?: typeof RoundedEnum;
  shadows?: typeof ShadowEnum;
  margins?: typeof MarginEnum;
  marginXs?: typeof MarginXEnum;
  marginYs?: typeof MarginYEnum;
  marginTops?: typeof MarginTopEnum;
  marginEnds?: typeof MarginEndEnum;
  marginBottoms?: typeof MarginBottomEnum;
  marginStarts?: typeof MarginStartEnum;
  paddings?: typeof PaddingEnum;
  paddingXs?: typeof PaddingXEnum;
  paddingYs?: typeof PaddingYEnum;
  paddingTops?: typeof PaddingTopEnum;
  paddingEnds?: typeof PaddingEndEnum;
  paddingBottoms?: typeof PaddingBottomEnum;
  paddingStarts?: typeof PaddingStartEnum;
  backgroundColors?: typeof BackgroundColorEnum;
  textColors?: typeof BoxTextColorEnum;
} = forwardRef(
  (
    {
      className,
      id,
      elementType = "div",
      children,
      disabled,
      border,
      borderColor,
      rounded,
      shadow,
      margin,
      marginX,
      marginY,
      marginTop,
      marginEnd,
      marginBottom,
      marginStart,
      padding,
      paddingX,
      paddingY,
      paddingTop,
      paddingEnd,
      paddingBottom,
      paddingStart,
      textColor,
      backgroundColor,
      scrollable,
      style
    }: BoxProps,
    ref
  ) => {
    const componentRef = useRef(null);
    const mergedRef = useMergeRef(ref, componentRef);
    const stylesTyped: Record<string, string> = styles;
    const getStyleByProp = (propValue: string): string | undefined =>
      propValue in stylesTyped ? stylesTyped[propValue] : undefined;

    return React.createElement(
      elementType,
      {
        ref: mergedRef,
        className: cx(
          styles.box,
          className,
          { [styles.opacityDisabled]: disabled, [styles.scrollable]: scrollable },
          getStyleByProp(border),
          getStyleByProp(borderColor),
          getStyleByProp(rounded),
          getStyleByProp(shadow),
          getStyleByProp(margin),
          getStyleByProp(marginX),
          getStyleByProp(marginY),
          getStyleByProp(marginTop),
          getStyleByProp(marginEnd),
          getStyleByProp(marginBottom),
          getStyleByProp(marginStart),
          getStyleByProp(padding),
          getStyleByProp(paddingX),
          getStyleByProp(paddingY),
          getStyleByProp(paddingTop),
          getStyleByProp(paddingEnd),
          getStyleByProp(paddingBottom),
          getStyleByProp(paddingStart),
          getStyleByProp(textColor),
          getStyleByProp(backgroundColor)
        ),
        id: id,
        style
      },
      children
    );
  }
);

export default withStaticProps(Box, {
  borders: BorderEnum,
  borderColors: BorderColorEnum,
  roundeds: RoundedEnum,
  shadows: ShadowEnum,
  margins: MarginEnum,
  marginXs: MarginXEnum,
  marginYs: MarginYEnum,
  marginTops: MarginTopEnum,
  marginEnds: MarginEndEnum,
  marginBottoms: MarginBottomEnum,
  marginStarts: MarginStartEnum,
  paddings: PaddingEnum,
  paddingXs: PaddingXEnum,
  paddingYs: PaddingYEnum,
  paddingTops: PaddingTopEnum,
  paddingEnds: PaddingEndEnum,
  paddingBottoms: PaddingBottomEnum,
  paddingStarts: PaddingStartEnum,
  backgroundColors: BackgroundColorEnum,
  textColors: BoxTextColorEnum
});
