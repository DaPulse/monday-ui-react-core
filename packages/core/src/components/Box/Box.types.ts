export type Disabled = "opacityDisabled";

export type Border = "border";

export type BorderColor = "borderColorUiBorderColor" | "borderColorLayoutBorderColor";

export type BackgroundColor =
  | "bgPrimaryBackgroundColor"
  | "bgSecondaryBackgroundColor"
  | "bgGreyBackgroundColor"
  | "bgAllgreyBackgroundColor"
  | "bgInvertedColorBackground";

export type BoxTextColor = "textPrimaryTextColor" | "textTextColorOnInverted" | "textSecondaryTextColor";

export type BoxPropSize = "xs" | "small" | "medium" | "large" | "xl" | "xxl" | "xxxl";

export type BoxPropSizeWithAuto = "auto" | BoxPropSize;

export type Shadow = Extract<BoxPropSize, "xs" | "small" | "medium" | "large">;

export type Rounded = "small" | "medium" | "big";
