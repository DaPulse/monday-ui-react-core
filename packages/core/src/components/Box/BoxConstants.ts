/**
 * @deprecated
 */
export enum Disabled {
  DISABLED = "opacityDisabled"
}

/**
 * @deprecated
 */
export enum Border {
  DEFAULT = "border"
}

/**
 * @deprecated
 */
export enum BorderColor {
  UI_BORDER_COLOR = "borderColorUiBorderColor",
  LAYOUT_BORDER_COLOR = "borderColorLayoutBorderColor"
}

/**
 * @deprecated
 */
export enum Rounded {
  SMALL = "roundedSmall",
  MEDIUM = "roundedMedium",
  BIG = "roundedBig"
}

/**
 * @deprecated
 */
export enum Shadow {
  XS = "shadowXs",
  SMALL = "shadowSmall",
  MEDIUM = "shadowMedium",
  LARGE = "shadowLarge"
}

/**
 * @deprecated
 */
export enum Margin {
  AUTO = "mAuto",
  XS = "mXs",
  SMALL = "mSmall",
  MEDIUM = "mMedium",
  LARGE = "mLarge",
  XL = "mXl",
  XXL = "mXxl",
  XXXL = "mXxxl"
}

/**
 * @deprecated
 */
export enum MarginX {
  AUTO = "mxAuto",
  XS = "mxXs",
  SMALL = "mxSmall",
  MEDIUM = "mxMedium",
  LARGE = "mxLarge",
  XL = "mxXl",
  XXL = "mxXxl",
  XXXL = "mxXxxl"
}

/**
 * @deprecated
 */
export enum MarginY {
  AUTO = "myAuto",
  XS = "myXs",
  SMALL = "mySmall",
  MEDIUM = "myMedium",
  LARGE = "myLarge",
  XL = "myXl",
  XXL = "myXxl",
  XXXL = "myXxxl"
}

/**
 * @deprecated
 */
export enum MarginTop {
  AUTO = "mtAuto",
  XS = "mtXs",
  SMALL = "mtSmall",
  MEDIUM = "mtMedium",
  LARGE = "mtLarge",
  XL = "mtXl",
  XXL = "mtXxl",
  XXXL = "mtXxxl"
}

/**
 * @deprecated
 */
export enum MarginEnd {
  AUTO = "meAuto",
  XS = "meXs",
  SMALL = "meSmall",
  MEDIUM = "meMedium",
  LARGE = "meLarge",
  XL = "meXl",
  XXL = "meXxl",
  XXXL = "meXxxl"
}

/**
 * @deprecated
 */
export enum MarginBottom {
  AUTO = "mbAuto",
  XS = "mbXs",
  SMALL = "mbSmall",
  MEDIUM = "mbMedium",
  LARGE = "mbLarge",
  XL = "mbXl",
  XXL = "mbXxl",
  XXXL = "mbXxxl"
}

/**
 * @deprecated
 */
export enum MarginStart {
  AUTO = "msAuto",
  XS = "msXs",
  SMALL = "msSmall",
  MEDIUM = "msMedium",
  LARGE = "msLarge",
  XL = "msXl",
  XXL = "msXxl",
  XXXL = "msXxxl"
}

/**
 * @deprecated
 */
export enum Padding {
  XS = "pXs",
  SMALL = "pSmall",
  MEDIUM = "pMedium",
  LARGE = "pLarge",
  XL = "pXl",
  XXL = "pXxl",
  XXXL = "pXxxl"
}

/**
 * @deprecated
 */
export enum PaddingX {
  XS = "pxXs",
  SMALL = "pxSmall",
  MEDIUM = "pxMedium",
  LARGE = "pxLarge",
  XL = "pxXl",
  XXL = "pxXxl",
  XXXL = "pxXxxl"
}

/**
 * @deprecated
 */
export enum PaddingY {
  XS = "pyXs",
  SMALL = "pySmall",
  MEDIUM = "pyMedium",
  LARGE = "pyLarge",
  XL = "pyXl",
  XXL = "pyXxl",
  XXXL = "pyXxxl"
}

/**
 * @deprecated
 */
export enum PaddingTop {
  XS = "ptXs",
  SMALL = "ptSmall",
  MEDIUM = "ptMedium",
  LARGE = "ptLarge",
  XL = "ptXl",
  XXL = "ptXxl",
  XXXL = "ptXxxl"
}

/**
 * @deprecated
 */
export enum PaddingEnd {
  XS = "peXs",
  SMALL = "peSmall",
  MEDIUM = "peMedium",
  LARGE = "peLarge",
  XL = "peXl",
  XXL = "peXxl",
  XXXL = "peXxxl"
}

/**
 * @deprecated
 */
export enum PaddingBottom {
  XS = "pbXs",
  SMALL = "pbSmall",
  MEDIUM = "pbMedium",
  LARGE = "pbLarge",
  XL = "pbXl",
  XXL = "pbXxl",
  XXXL = "pbXxxl"
}

/**
 * @deprecated
 */
export enum PaddingStart {
  XS = "psXs",
  SMALL = "psSmall",
  MEDIUM = "psMedium",
  LARGE = "psLarge",
  XL = "psXl",
  XXL = "psXxl",
  XXXL = "psXxxl"
}

/**
 * @deprecated
 */
export enum BackgroundColor {
  PRIMARY_BACKGROUND_COLOR = "bgPrimaryBackgroundColor",
  SECONDARY_BACKGROUND_COLOR = "bgSecondaryBackgroundColor",
  GREY_BACKGROUND_COLOR = "bgGreyBackgroundColor",
  ALL_GREY_BACKGROUND_COLOR = "bgAllgreyBackgroundColor",
  INVERTED_COLOR_BACKGROUND = "bgInvertedColorBackground"
}

/**
 * @deprecated
 */
export enum BoxTextColor {
  PRIMARY_TEXT_COLOR = "textPrimaryTextColor",
  TEXT_COLOR_ON_INVERTED = "textTextColorOnInverted",
  SECONDARY_TEXT_COLOR = "textSecondaryTextColor"
}

export const SizePrefixMapping = {
  margin: "m",
  marginX: "mx",
  marginY: "my",
  marginTop: "mt",
  marginEnd: "me",
  marginBottom: "mb",
  marginStart: "ms",
  padding: "p",
  paddingX: "px",
  paddingY: "py",
  paddingTop: "pt",
  paddingEnd: "pe",
  paddingBottom: "pb",
  paddingStart: "ps"
};
