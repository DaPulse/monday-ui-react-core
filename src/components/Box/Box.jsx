import React, { useRef, forwardRef } from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import useMergeRefs from "hooks/useMergeRefs";
import styles from "./Box.module.scss";
import {
  DISABLED,
  BORDER,
  BORDER_TOP,
  BORDER_END,
  BORDER_BOTTOM,
  BORDER_START,
  BORDER_COLOR,
  ROUNDED,
  ROUNDED_TOP,
  ROUNDED_END,
  ROUNDED_BOTTOM,
  ROUNDED_START,
  SHADOW,
  MARGIN,
  MARGIN_X,
  MARGIN_Y,
  MARGIN_TOP,
  MARGIN_END,
  MARGIN_BOTTOM,
  MARGIN_START,
  PADDING,
  PADDING_X,
  PADDING_Y,
  PADDING_TOP,
  PADDING_END,
  PADDING_BOTTOM,
  PADDING_START,
  BACKGROUND_COLORS,
  COLORS
} from "./BoxConstants";

const Box = forwardRef(
  (
    {
      className,
      id,
      elementType,
      children,
      disabled,
      border,
      borderTop,
      borderEnd,
      borderBottom,
      borderStart,
      borderColor,
      rounded,
      roundedTop,
      roundedEnd,
      roundedBottom,
      roundedStart,
      shadow,
      margin,
      marginX,
      marginY,
      marginTop,
      marginEnd,
      marginBottom,
      marginStart,
      padding,
      paddingX,
      paddingY,
      paddingTop,
      paddingEnd,
      paddingBottom,
      paddingStart,
      textColor,
      backgroundColor
    },
    ref
  ) => {
    const componentRef = useRef(null);
    const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
    const Element = elementType;
    return (
      <Element
        ref={mergedRef}
        className={cx(
          styles.box,
          className,
          { [DISABLED.DISABLED]: disabled },
          border,
          borderTop,
          borderEnd,
          borderBottom,
          borderStart,
          borderColor,
          rounded,
          roundedTop,
          roundedEnd,
          roundedBottom,
          roundedStart,
          shadow,
          margin,
          marginX,
          marginY,
          marginTop,
          marginEnd,
          marginBottom,
          marginStart,
          padding,
          paddingX,
          paddingY,
          paddingTop,
          paddingEnd,
          paddingBottom,
          paddingStart,
          textColor,
          backgroundColor
        )}
        id={id}
      >
        {children}
      </Element>
    );
  }
);

Box.borders = BORDER;
Box.borderTops = BORDER_TOP;
Box.borderEnds = BORDER_END;
Box.borderBottoms = BORDER_BOTTOM;
Box.borderStarts = BORDER_START;
Box.borderColors = BORDER_COLOR;
Box.roundeds = ROUNDED;
Box.roundedTops = ROUNDED_TOP;
Box.roundedEnds = ROUNDED_END;
Box.roundedBottoms = ROUNDED_BOTTOM;
Box.roundedStarts = ROUNDED_START;
Box.shadows = SHADOW;
Box.margins = MARGIN;
Box.marginXs = MARGIN_X;
Box.marginYs = MARGIN_Y;
Box.marginTops = MARGIN_TOP;
Box.marginEnds = MARGIN_END;
Box.marginBottoms = MARGIN_BOTTOM;
Box.marginStarts = MARGIN_START;
Box.paddings = PADDING;
Box.paddingXs = PADDING_X;
Box.paddingYs = PADDING_Y;
Box.paddingTops = PADDING_TOP;
Box.paddingEnds = PADDING_END;
Box.paddingBottoms = PADDING_BOTTOM;
Box.paddingStarts = PADDING_START;
Box.backgroundColors = BACKGROUND_COLORS;
Box.textColors = COLORS;

Box.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string,
  elementType: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element)
  ]),
  disabled: PropTypes.bool,
  border: PropTypes.oneOf([Box.borders.NONE, Box.borders.DEFAULT]),
  borderTop: PropTypes.oneOf([Box.borderTops.NONE, Box.borderTops.DEFAULT]),
  borderEnd: PropTypes.oneOf([Box.borderEnds.NONE, Box.borderEnds.DEFAULT]),
  borderBottom: PropTypes.oneOf([Box.borderBottoms.NONE, Box.borderBottoms.DEFAULT]),
  borderStart: PropTypes.oneOf([Box.borderStarts.NONE, Box.borderStarts.DEFAULT]),
  borderColor: PropTypes.oneOf([Box.borderColors.UI_BORDER_COLOR, Box.borderColors.LAYOUT_BORDER_COLOR]),
  rounded: PropTypes.oneOf([Box.roundeds.NONE, Box.roundeds.SMALL, Box.roundeds.MEDIUM, Box.roundeds.BIG]),
  roundedTop: PropTypes.oneOf([
    Box.roundedTops.NONE,
    Box.roundedTops.SMALL,
    Box.roundedTops.MEDIUM,
    Box.roundedTops.BIG
  ]),
  roundedEnd: PropTypes.oneOf([
    Box.roundedEnds.NONE,
    Box.roundedEnds.SMALL,
    Box.roundedEnds.MEDIUM,
    Box.roundedEnds.BIG
  ]),
  roundedBottom: PropTypes.oneOf([
    Box.roundedBottoms.NONE,
    Box.roundedBottoms.SMALL,
    Box.roundedBottoms.MEDIUM,
    Box.roundedBottoms.BIG
  ]),
  roundedStart: PropTypes.oneOf([
    Box.roundedStarts.NONE,
    Box.roundedStarts.SMALL,
    Box.roundedStarts.MEDIUM,
    Box.roundedStarts.BIG
  ]),
  shadow: PropTypes.oneOf([Box.shadows.XS, Box.shadows.SMALL, Box.shadows.MEDIUM, Box.shadows.LARGE]),
  margin: PropTypes.oneOf([
    Box.margins.NONE,
    Box.margins.AUTO,
    Box.margins.XS,
    Box.margins.SMALL,
    Box.margins.MEDIUM,
    Box.margins.LARGE,
    Box.margins.XL,
    Box.margins.XXL,
    Box.margins.XXXL
  ]),
  marginX: PropTypes.oneOf([
    Box.marginXs.NONE,
    Box.marginXs.AUTO,
    Box.marginXs.XS,
    Box.marginXs.SMALL,
    Box.marginXs.MEDIUM,
    Box.marginXs.LARGE,
    Box.marginXs.XL,
    Box.marginXs.XXL,
    Box.marginXs.XXXL
  ]),
  marginY: PropTypes.oneOf([
    Box.marginYs.NONE,
    Box.marginYs.AUTO,
    Box.marginYs.XS,
    Box.marginYs.SMALL,
    Box.marginYs.MEDIUM,
    Box.marginYs.LARGE,
    Box.marginYs.XL,
    Box.marginYs.XXL,
    Box.marginYs.XXXL
  ]),
  marginTop: PropTypes.oneOf([
    Box.marginTops.NONE,
    Box.marginTops.AUTO,
    Box.marginTops.XS,
    Box.marginTops.SMALL,
    Box.marginTops.MEDIUM,
    Box.marginTops.LARGE,
    Box.marginTops.XL,
    Box.marginTops.XXL,
    Box.marginTops.XXXL
  ]),
  marginEnd: PropTypes.oneOf([
    Box.marginEnds.NONE,
    Box.marginEnds.AUTO,
    Box.marginEnds.XS,
    Box.marginEnds.SMALL,
    Box.marginEnds.MEDIUM,
    Box.marginEnds.LARGE,
    Box.marginEnds.XL,
    Box.marginEnds.XXL,
    Box.marginEnds.XXXL
  ]),
  marginBottom: PropTypes.oneOf([
    Box.marginBottoms.NONE,
    Box.marginBottoms.AUTO,
    Box.marginBottoms.XS,
    Box.marginBottoms.SMALL,
    Box.marginBottoms.MEDIUM,
    Box.marginBottoms.LARGE,
    Box.marginBottoms.XL,
    Box.marginBottoms.XXL,
    Box.marginBottoms.XXXL
  ]),
  marginStart: PropTypes.oneOf([
    Box.marginStarts.NONE,
    Box.marginStarts.AUTO,
    Box.marginStarts.XS,
    Box.marginStarts.SMALL,
    Box.marginStarts.MEDIUM,
    Box.marginStarts.LARGE,
    Box.marginStarts.XL,
    Box.marginStarts.XXL,
    Box.marginStarts.XXXL
  ]),
  padding: PropTypes.oneOf([
    Box.paddings.NONE,
    Box.paddings.XS,
    Box.paddings.SMALL,
    Box.paddings.MEDIUM,
    Box.paddings.LARGE,
    Box.paddings.XL,
    Box.paddings.XXL,
    Box.paddings.XXXL
  ]),
  paddingX: PropTypes.oneOf([
    Box.paddingXs.NONE,
    Box.paddingXs.XS,
    Box.paddingXs.SMALL,
    Box.paddingXs.MEDIUM,
    Box.paddingXs.LARGE,
    Box.paddingXs.XL,
    Box.paddingXs.XXL,
    Box.paddingXs.XXXL
  ]),
  paddingY: PropTypes.oneOf([
    Box.paddingYs.NONE,
    Box.paddingYs.XS,
    Box.paddingYs.SMALL,
    Box.paddingYs.MEDIUM,
    Box.paddingYs.LARGE,
    Box.paddingYs.XL,
    Box.paddingYs.XXL,
    Box.paddingYs.XXXL
  ]),
  paddingTop: PropTypes.oneOf([
    Box.paddingTops.NONE,
    Box.paddingTops.XS,
    Box.paddingTops.SMALL,
    Box.paddingTops.MEDIUM,
    Box.paddingTops.LARGE,
    Box.paddingTops.XL,
    Box.paddingTops.XXL,
    Box.paddingTops.XXXL
  ]),
  paddingEnd: PropTypes.oneOf([
    Box.paddingEnds.NONE,
    Box.paddingEnds.XS,
    Box.paddingEnds.SMALL,
    Box.paddingEnds.MEDIUM,
    Box.paddingEnds.LARGE,
    Box.paddingEnds.XL,
    Box.paddingEnds.XXL,
    Box.paddingEnds.XXXL
  ]),
  paddingBottom: PropTypes.oneOf([
    Box.paddingBottoms.NONE,
    Box.paddingBottoms.XS,
    Box.paddingBottoms.SMALL,
    Box.paddingBottoms.MEDIUM,
    Box.paddingBottoms.LARGE,
    Box.paddingBottoms.XL,
    Box.paddingBottoms.XXL,
    Box.paddingBottoms.XXXL
  ]),
  paddingStart: PropTypes.oneOf([
    Box.paddingStarts.NONE,
    Box.paddingStarts.XS,
    Box.paddingStarts.SMALL,
    Box.paddingStarts.MEDIUM,
    Box.paddingStarts.LARGE,
    Box.paddingStarts.XL,
    Box.paddingStarts.XXL,
    Box.paddingStarts.XXXL
  ]),
  textColor: PropTypes.oneOf([
    Box.textColors.PRIMARY_COLOR,
    Box.textColors.PRIMARY_HOVER_COLOR,
    Box.textColors.PRIMARY_SELECTED_COLOR,
    Box.textColors.PRIMARY_TEXT_COLOR,
    Box.textColors.TEXT_COLOR_ON_PRIMARY,
    Box.textColors.TEXT_COLOR_ON_INVERTED,
    Box.textColors.SECONDARY_TEXT_COLOR,
    Box.textColors.PLACEHOLDER_COLOR,
    Box.textColors.ICON_COLOR,
    Box.textColors.LINK_COLOR,
    Box.textColors.DISABLED_TEXT_COLOR
  ]),
  backgroundColor: PropTypes.oneOf([
    Box.backgroundColors.PRIMARY_BACKGROUND_COLOR,
    Box.backgroundColors.PRIMARY_HOVER_BACKGROUND_COLOR,
    Box.backgroundColors.SECONDARY_BACKGROUND_COLOR,
    Box.backgroundColors.GREY_BACKGROUND_COLOR,
    Box.backgroundColors.ALL_GREY_BACKGROUND_COLOR,
    Box.backgroundColors.INVERTED_COLOR_BACKGROUND,
    Box.backgroundColors.DISABLED_BACKGROUND_COLOR
  ])
};
Box.defaultProps = {
  className: "",
  id: undefined,
  elementType: "div",
  children: undefined,
  disabled: undefined,
  border: undefined,
  borderTop: undefined,
  borderEnd: undefined,
  borderBottom: undefined,
  borderStart: undefined,
  borderColor: undefined,
  rounded: undefined,
  roundedTop: undefined,
  roundedEnd: undefined,
  roundedBottom: undefined,
  roundedStart: undefined,
  shadow: undefined,
  margin: undefined,
  marginX: undefined,
  marginY: undefined,
  marginTop: undefined,
  marginEnd: undefined,
  marginBottom: undefined,
  marginStart: undefined,
  padding: undefined,
  paddingX: undefined,
  paddingY: undefined,
  paddingTop: undefined,
  paddingEnd: undefined,
  paddingBottom: undefined,
  paddingStart: undefined,
  textColor: undefined,
  backgroundColor: undefined
};

export default Box;
