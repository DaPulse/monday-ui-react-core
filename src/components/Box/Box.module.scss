@import "../../styles/themes";

// Monday ui styles dependencies
@import "~monday-ui-style/src/common";

// Utilities map
@import "../../styles/utilities/utilities-settings";


// Generating Utility class locally for Box component
@include utility-class-factory($utilities-modules, $export: "modules");

