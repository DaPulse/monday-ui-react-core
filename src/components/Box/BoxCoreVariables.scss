// BOX Core: variables
// Mock Custom properties that should be imported from "monday-ui-styles"

:root {
  --border-width: 1px;
  --border-style: solid;
}

:root,
.light-app-theme,
.default-app-theme {
  --box-shadow-xs: 0px 4px 6px -4px rgba(0, 0, 0, 0.1);
}

.dark-app-theme {
  --box-shadow-xs: 0px 4px 6px -4px rgba(9, 11, 25, 0.5);
}

.black-app-theme {
  --box-shadow-xs: 0px 4px 6px -4px #000000;
}

.hacker-app-theme {
  --box-shadow-xs: 0px 4px 6px -4px #000000;
}