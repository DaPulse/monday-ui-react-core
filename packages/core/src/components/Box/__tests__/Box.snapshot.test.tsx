import React from "react";
import renderer from "react-test-renderer";
import Box from "../Box";

jest.mock("../Box.module.scss", () => ({
  box: "box",
  border: "border",
  borderColorUiBorderColor: "borderColorUiBorderColor",
  roundedSmall: "roundedSmall",
  shadowSmall: "shadowSmall",
  mSmall: "mSmall",
  mxSmall: "mxSmall",
  mySmall: "mySmall",
  mtSmall: "mtSmall",
  meSmall: "meSmall",
  msSmall: "msSmall",
  pSmall: "pSmall",
  pxSmall: "pxSmall",
  pySmall: "pySmall",
  ptSmall: "ptSmall",
  peSmall: "peSmall",
  pbSmall: "pbSmall",
  psSmall: "psSmall",
  bgPrimaryBackgroundColor: "bgPrimaryBackgroundColor",
  textPrimaryTextColor: "textPrimaryTextColor"
}));

describe("Box renders correctly", () => {
  it("with empty props", () => {
    const tree = renderer.create(<Box></Box>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with children", () => {
    const tree = renderer.create(<Box>Child</Box>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with border prop", () => {
    const tree = renderer.create(<Box border="border" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with borderColor prop", () => {
    const tree = renderer.create(<Box borderColor="borderColorUiBorderColor" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with rounded prop", () => {
    const tree = renderer.create(<Box rounded="roundedSmall" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with shadow prop", () => {
    const tree = renderer.create(<Box shadow="shadowSmall" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with margin prop", () => {
    const tree = renderer.create(<Box margin="mSmall" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with marginX prop", () => {
    const tree = renderer.create(<Box marginX="mxSmall" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with marginY prop", () => {
    const tree = renderer.create(<Box marginY="mySmall" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with marginTop prop", () => {
    const tree = renderer.create(<Box marginTop="mtSmall" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with marginEnd prop", () => {
    const tree = renderer.create(<Box marginEnd="meSmall" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with marginStart prop", () => {
    const tree = renderer.create(<Box marginStart="msSmall" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with padding prop", () => {
    const tree = renderer.create(<Box padding="pSmall" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with paddingX prop", () => {
    const tree = renderer.create(<Box paddingX="pxSmall" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with paddingY prop", () => {
    const tree = renderer.create(<Box paddingY="pySmall" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with paddingTop prop", () => {
    const tree = renderer.create(<Box paddingTop="ptSmall" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with paddingEnd prop", () => {
    const tree = renderer.create(<Box paddingEnd="peSmall" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with paddingBottom prop", () => {
    const tree = renderer.create(<Box paddingBottom="pbSmall" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with paddingStart prop", () => {
    const tree = renderer.create(<Box paddingStart="psSmall" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with backgroundColor prop", () => {
    const tree = renderer.create(<Box backgroundColor="bgPrimaryBackgroundColor" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with text prop", () => {
    const tree = renderer.create(<Box textColor="textPrimaryTextColor" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
