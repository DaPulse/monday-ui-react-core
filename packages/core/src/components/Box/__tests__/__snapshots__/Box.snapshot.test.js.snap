// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders correctly with backgronudColor prop 1`] = `
<div
  className="box scrollable"
/>
`;

exports[`Box renders correctly with border prop 1`] = `
<div
  className="box scrollable border"
/>
`;

exports[`Box renders correctly with borderColor prop 1`] = `
<div
  className="box scrollable borderColorUiBorderColor"
/>
`;

exports[`Box renders correctly with children 1`] = `
<div
  className="box scrollable"
>
  Child
</div>
`;

exports[`Box renders correctly with empty props 1`] = `
<div
  className="box scrollable"
/>
`;

exports[`Box renders correctly with margin prop 1`] = `
<div
  className="box scrollable mSmall"
/>
`;

exports[`Box renders correctly with marginEnd prop 1`] = `
<div
  className="box scrollable meSmall"
/>
`;

exports[`Box renders correctly with marginStart prop 1`] = `
<div
  className="box scrollable msSmall"
/>
`;

exports[`Box renders correctly with marginTop prop 1`] = `
<div
  className="box scrollable mtSmall"
/>
`;

exports[`Box renders correctly with marginX prop 1`] = `
<div
  className="box scrollable mxSmall"
/>
`;

exports[`Box renders correctly with marginY prop 1`] = `
<div
  className="box scrollable mySmall"
/>
`;

exports[`Box renders correctly with padding prop 1`] = `
<div
  className="box scrollable pSmall"
/>
`;

exports[`Box renders correctly with paddingBottom prop 1`] = `
<div
  className="box scrollable pbSmall"
/>
`;

exports[`Box renders correctly with paddingEnd prop 1`] = `
<div
  className="box scrollable peSmall"
/>
`;

exports[`Box renders correctly with paddingStart prop 1`] = `
<div
  className="box scrollable psSmall"
/>
`;

exports[`Box renders correctly with paddingTop prop 1`] = `
<div
  className="box scrollable ptSmall"
/>
`;

exports[`Box renders correctly with paddingX prop 1`] = `
<div
  className="box scrollable pxSmall"
/>
`;

exports[`Box renders correctly with paddingY prop 1`] = `
<div
  className="box scrollable pySmall"
/>
`;

exports[`Box renders correctly with rounded prop 1`] = `
<div
  className="box scrollable roundedSmall"
/>
`;

exports[`Box renders correctly with shadow prop 1`] = `
<div
  className="box scrollable shadowSmall"
/>
`;

exports[`Box renders correctly with text prop 1`] = `
<div
  className="box scrollable textPrimaryTextColor"
/>
`;
